src/widgetastic_patternfly4/button.py

```diff
@@ -39,46 +37,60 @@ class Button(Widget, ClickableMixin):
-        Widget.__init__(self, parent, logger=logger)
-        self.args = text
-        self.kwargs = kwargs
+    def _generate_locator(self, *text, **kwargs):
         classes = kwargs.pop("classes", [])
         if text:
             if kwargs:  # classes should have been the only kwarg combined with text args
                 raise TypeError("If you pass button text then only pass classes in addition")
             if len(text) == 1:
-                self.locator_conditions = "normalize-space(.)={}".format(quote(text[0]))
+                locator_conditions = "normalize-space(.)={}".format(quote(text[0]))
             elif len(text) == 2 and text[0].lower() == "contains":
-                self.locator_conditions = "contains(normalize-space(.), {})".format(quote(text[1]))
+                locator_conditions = "contains(normalize-space(.), {})".format(quote(text[1]))
             else:
-                raise TypeError("An illegal combination of text params")
+                raise TypeError("An illegal combination of args/kwargs")
         else:
             # Join the kwargs, if any
-            self.locator_conditions = " and ".join(
+            locator_conditions = " and ".join(
                 "@{}={}".format(attr, quote(value)) for attr, value in kwargs.items()
             )
 
         if classes:
-            if self.locator_conditions:
-                self.locator_conditions += " and "
-            self.locator_conditions += " and ".join(
+            if locator_conditions:
+                locator_conditions += " and "
+            locator_conditions += " and ".join(
                 "contains(@class, {})".format(quote(klass)) for klass in classes
             )
-        if self.locator_conditions:
-            self.locator_conditions = "and ({})".format(self.locator_conditions)
+        if locator_conditions:
+            locator_conditions = "and ({})".format(locator_conditions)
+
+        return (
+            ".//*[(self::a or self::button or (self::input and "
+            "(@type='button' or @type='submit'))) and "
+            f"contains(@class, 'pf-c-button') {locator_conditions}]"
+        )
+
+    def __init__(self, parent, *text, **kwargs):
+        logger = kwargs.pop("logger", None)
+        Widget.__init__(self, parent, logger=logger)
+        self.args = text
+        self.kwargs = kwargs
+        locator = kwargs.pop("locator", None)
+        if locator:
+            self.locator = locator
+        else:
+            self.locator = self._generate_locator(*text, **kwargs)
+
+    def read(self):
+        return self.browser.text(self)
 
     @property
     def active(self):
         return "pf-m-active" in self.browser.classes(self)
 
     @property
     def disabled(self):
-        return (
-            "pf-m-disabled" in self.browser.classes(self) or
-            self.browser.get_attribute("disabled", self) == "disabled"
-        )
+        check1 = "pf-m-disabled" in self.browser.classes(self)
```

Comment:
Note this change is just because flake8 was complaining about line breaks after the `or`